packages/network-controller/src/NetworkController.ts

```diff
@@ -1063,205 +1482,427 @@ export class NetworkController extends BaseController<
-    );
-    const { rpcUrl, chainId, ticker } = sanitizedNetworkConfiguration;
+      replacementSelectedRpcEndpointIndex,
+    }: { replacementSelectedRpcEndpointIndex?: number } = {},
+  ): Promise<NetworkConfiguration> {
+    const existingNetworkConfiguration =
+      this.state.networkConfigurationsByChainId[chainId];
 
-    assertIsStrictHexString(chainId);
-    if (!isSafeChainId(chainId)) {
+    if (existingNetworkConfiguration === undefined) {
       throw new Error(
-        `Invalid chain ID "${chainId}": numerical value greater than max safe value.`,
-      );
-    }
-    if (!rpcUrl) {
-      throw new Error(
-        'An rpcUrl is required to add or update network configuration',
-      );
-    }
-    if (!referrer || !source) {
-      throw new Error(
-        'referrer and source are required arguments for adding or updating a network configuration',
-      );
-    }
-    try {
-      new URL(rpcUrl);
-      // TODO: Replace `any` with type
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    } catch (e: any) {
-      if (e.message.includes('Invalid URL')) {
-        throw new Error('rpcUrl must be a valid URL');
-      }
-    }
-    if (!ticker) {
-      throw new Error(
-        'A ticker is required to add or update networkConfiguration',
+        `Could not update network: Cannot find network configuration for chain ${inspect(
+          chainId,
+        )}`,
       );
     }
 
+    const existingChainId = chainId;
+    const { chainId: newChainId, rpcEndpoints: setOfNewRpcEndpointFields } =
+      fields;
+
     const autoManagedNetworkClientRegistry =
       this.#ensureAutoManagedNetworkClientRegistryPopulated();
 
-    const existingNetworkConfiguration = Object.values(
-      this.state.networkConfigurations,
-    ).find(
-      (networkConfig) =>
-        networkConfig.rpcUrl.toLowerCase() === rpcUrl.toLowerCase(),
-    );
-    const upsertedNetworkConfigurationId = existingNetworkConfiguration
-      ? existingNetworkConfiguration.id
-      : random();
-    const networkClientId = upsertedNetworkConfigurationId;
-
-    const customNetworkClientRegistry =
-      autoManagedNetworkClientRegistry[NetworkClientType.Custom];
-    const existingAutoManagedNetworkClient =
-      customNetworkClientRegistry[networkClientId];
-    const shouldDestroyExistingNetworkClient =
-      existingAutoManagedNetworkClient &&
-      existingAutoManagedNetworkClient.configuration.chainId !== chainId;
-    if (shouldDestroyExistingNetworkClient) {
-      existingAutoManagedNetworkClient.destroy();
+    this.#validateNetworkFields({
+      mode: 'update',
+      errorMessagePrefix: 'Could not update network',
+      networkFields: fields,
+      existingNetworkConfiguration,
+      autoManagedNetworkClientRegistry,
+    });
+
+    const networkClientOperations: NetworkClientOperation[] = [];
+
+    for (const newRpcEndpointFields of setOfNewRpcEndpointFields) {
+      const existingRpcEndpointForNoop =
```

Comment:
The logic seems sound but as you mentioned it's quite bulky - I agree that moving these `CRUD` operations on Networks would help a lot, especially because we could then move all config validators to a more specific class, making `NetworkController` simpler